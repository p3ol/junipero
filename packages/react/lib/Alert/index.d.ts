import {
  ReactNode,
  ElementType,
  MutableRefObject,
  ComponentPropsWithRef,
} from 'react';

export declare type AlertRef = {
  isJunipero: boolean;
  innerRef: MutableRefObject<any>;
};

export declare interface AlertProps extends ComponentPropsWithRef<any> {
  animationTimeout?: number;
  children?: ReactNode | JSX.Element;
  className?: string;
  icon?: ReactNode | JSX.Element;
  index?: string | number;
  lifespan?: number;
  tag?: string | ElementType;
  title?: ReactNode | JSX.Element;
  animate?(
    alert: ReactNode | JSX.Element,
    opts: { opened: boolean; index: string | number }
  ): ReactNode | JSX.Element;
  onDismiss?(index?: string | number): void;
  ref?: MutableRefObject<AlertRef | undefined>;
}

export declare interface AlertObject {
  animationTimeout?: number;
  content: ReactNode | JSX.Element;
  icon?: ReactNode | JSX.Element;
  index?: string | number;
  title?: ReactNode | JSX.Element;
  type?: string;
  duration?: number;
  lifespan?: number;
  animate?(
    alert: ReactNode | JSX.Element,
    opts: { opened: boolean; index: string | number }
  ): ReactNode | JSX.Element;
  onDismiss?(index?: string | number): void;
}

declare function Alert(props: AlertProps): ReactNode | JSX.Element;

export default Alert;
