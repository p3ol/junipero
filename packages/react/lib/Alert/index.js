import { forwardRef, useImperativeHandle, useRef } from 'react';
import { classNames } from '@junipero/core';
import { useTimeout } from '@junipero/hooks';
import PropTypes from 'prop-types';

import { useAlerts } from '../hooks';
import { Remove } from '../icons';
import Card from '../Card';

const Alert = forwardRef(({
  className,
  index,
  icon,
  title,
  lifespan,
  tag: Tag = 'div',
  children,
  onDismiss,
  ...rest
}, ref) => {
  const innerRef = useRef();
  const { dismiss } = useAlerts();

  useTimeout(() => {
    dismiss?.(index);
  }, lifespan || 5000, [], { enabled: !!lifespan });

  useImperativeHandle(ref, () => ({
    innerRef,
    isJunipero: true,
  }));

  const onClose_ = e => {
    e.preventDefault();
    dismiss?.(index);
    onDismiss?.();
  };

  return (
    <Tag
      { ...rest }
      ref={ref}
      className={classNames('junipero', 'alert', className)}
      onClick={onClose_}
    >
      <Card>
        { icon && (
          <div className="type-icon">{ icon }</div>
        ) }
        <div className="content">
          <h6 className="junipero">{ title }</h6>
          <div className="junipero secondary">{ children }</div>
        </div>
        <Remove className="close" />
      </Card>
    </Tag>
  );
});

Alert.displayName = 'Alert';
Alert.propTypes = {
  index: PropTypes.any,
  icon: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.string,
    PropTypes.func,
    PropTypes.object,
  ]),
  lifespan: PropTypes.number,
  tag: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.string,
    PropTypes.func,
    PropTypes.object,
  ]),
  title: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.string,
    PropTypes.func,
    PropTypes.object,
  ]),
  onDismiss: PropTypes.func,
};

export default Alert;
