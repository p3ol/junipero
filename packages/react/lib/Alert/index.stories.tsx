import Alert from './index';

export default { title: 'react/Alert' };

const icon = <i className="material-icons">info</i>;

export const basic = () =>
  ['', 'primary', 'success', 'warning', 'danger'].map(t => (
    <Alert key={t} icon={icon} title="Title" { ...t && { className: t }}>
      This is an alert
    </Alert>
  ));
