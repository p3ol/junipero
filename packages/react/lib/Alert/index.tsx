import {
  type MutableRefObject,
  type ReactNode,
  type ComponentPropsWithRef,
  type ElementType,
  forwardRef,
  useState,
  useImperativeHandle,
  useRef,
} from 'react';
import { type ForwardedProps, classNames } from '@junipero/core';
import { useTimeout } from '@junipero/hooks';
import PropTypes from 'prop-types';

import { Remove } from '../icons';
import Card from '../Card';

export declare type AlertRef = {
  isJunipero: boolean;
  innerRef: MutableRefObject<any>;
};

export declare interface AlertProps extends ComponentPropsWithRef<any>{
  animationTimeout?: number;
  children?: ReactNode | JSX.Element;
  className?: string;
  icon?: ReactNode | JSX.Element;
  index?: string | number;
  lifespan?: number;
  tag?: string | ElementType;
  animate?(
    alert: ReactNode | JSX.Element,
    opts: { opened: boolean; index: string | number }
  ): ReactNode | JSX.Element;
  onDismiss?(index?: string | number): void;
  ref?: MutableRefObject<AlertRef | undefined>;
  onClick?: (e: Event) => void;
  title?: ReactNode | JSX.Element;
}

export declare interface AlertObject{
  id?: number | string,
  animationTimeout?: number;
  content?: ReactNode | JSX.Element;
  icon?: ReactNode | JSX.Element;
  index?: string | number;
  title?: ReactNode | JSX.Element;
  duration?: number;
  type?: string;
  lifespan?: number;
  animate?(
    alert: ReactNode | JSX.Element,
    opts: { opened: boolean; index: string | number }
  ): ReactNode | JSX.Element;
  onDismiss?(index?: string | number): void;
}

const Alert = forwardRef(({
  animationTimeout = 100,
  tag: Tag = 'div',
  lifespan = 0,
  index,
  animate,
  className,
  icon,
  title,
  children,
  onClick,
  onDismiss,
  ...rest
}: AlertProps, ref) => {
  const innerRef = useRef();
  const [enabled, setEnabled] = useState<boolean>(true);
  const timeout = animate ? animationTimeout : 0;

  useTimeout(() => {
    setEnabled(false);
  }, lifespan + timeout, [lifespan], { enabled: lifespan > 0 });

  useTimeout(() => {
    onDismiss?.(index);
  }, timeout, [enabled], { enabled: !enabled });

  useImperativeHandle(ref, () => ({
    innerRef,
    isJunipero: true,
  }));

  const onClick_ = (e : Event) => {
    onClick?.(e);
    setEnabled(false);
  };

  const content = (
    <Tag
      { ...rest }
      ref={ref}
      className={classNames('junipero', 'alert', className)}
      onClick={onClick_}
    >
      <Card>
        { icon && (
          <div className="type-icon">{ icon }</div>
        ) }
        <div className="content">
          <h6 className="junipero">{ title }</h6>
          <div className="junipero secondary">{ children }</div>
        </div>
        <Remove className="close" />
      </Card>
    </Tag>
  );

  return animate ? animate(content, { opened: enabled, index }) : content;
}) as ForwardedProps<AlertProps, AlertRef>;

Alert.displayName = 'Alert';
Alert.propTypes = {
  animate: PropTypes.func,
  animationTimeout: PropTypes.number,
  index: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),
  icon: PropTypes.oneOfType([
    PropTypes.node,
  ]),
  lifespan: PropTypes.number,
  tag: PropTypes.any, //TODO fixme
  title: PropTypes.any,
  onDismiss: PropTypes.func,
  onClick: PropTypes.func,
};

export default Alert;
