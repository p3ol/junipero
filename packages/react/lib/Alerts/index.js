import { forwardRef, useImperativeHandle, useRef } from 'react';
import { classNames } from '@junipero/core';
import PropTypes from 'prop-types';

import { useAlerts } from '../hooks';
import Alert from '../Alert';

const Alerts = forwardRef(({ className, animate, ...rest }, ref) => {
  const innerRef = useRef();
  const { alerts } = useAlerts();

  useImperativeHandle(ref, () => ({
    innerRef,
    alerts,
    isJunipero: true,
  }));

  const anim = animate || (a => a);

  return (
    <div
      {...rest}
      ref={innerRef}
      className={classNames('junipero', 'alerts', className)}
    >
      { alerts?.map((alert, index) => anim((
        <Alert
          key={index}
          index={index}
          icon={alert.icon}
          title={alert.title}
          lifespan={alert.duration ?? alert.lifespan}
          onDismiss={alert.onDismiss}
          className={alert.type}
        >
          { alert.content }
        </Alert>
      ), index)) }
    </div>
  );
});

Alerts.displayName = 'Alerts';
Alerts.propTypes = {
  animate: PropTypes.func,
};

export default Alerts;
