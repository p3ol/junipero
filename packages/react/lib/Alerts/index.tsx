import {
  type ComponentPropsWithoutRef,
  type MutableRefObject,
  type ReactNode,
  forwardRef,
  useImperativeHandle,
  useRef,
} from 'react';
import { classNames, omit } from '@junipero/core';

import type { JuniperoRef } from '../types';
import type { TransitionProps } from '../Transition';
import { useAlerts } from '../hooks';
import Alert, { type AlertObject } from '../Alert';

export declare interface AlertsRef extends JuniperoRef {
  alerts: Array<AlertObject>;
  innerRef: MutableRefObject<HTMLDivElement>;
}

export declare type AlertsTypes =
  | 'danger'
  | 'default'
  | 'primary'
  | 'success'
  | 'warning';

export declare interface AlertsProps extends ComponentPropsWithoutRef<'div'> {
  animationTimeout?: number;
  icons?: {
    [type in AlertsTypes]?: ReactNode | JSX.Element;
  };
  animateAlert?(
    alert: ReactNode | JSX.Element,
    opts: {
      opened: boolean;
      index: string | number;
    } & Partial<TransitionProps>,
  ): ReactNode | JSX.Element;
}

const Alerts = forwardRef<AlertsRef, AlertsProps>(({
  className,
  animateAlert,
  animationTimeout,
  icons,
  ...rest
}: AlertsProps, ref) => {
  const innerRef = useRef<HTMLDivElement>();
  const { alerts, dismiss } = useAlerts();

  useImperativeHandle(ref, () => ({
    innerRef,
    alerts,
    isJunipero: true,
  }));

  const onDismiss = (alert: AlertObject, index: number) => {
    dismiss?.(alert);
    alert?.onDismiss?.(index);
  };

  return (
    <div
      {...rest}
      ref={innerRef}
      className={classNames('junipero', 'alerts', className)}
    >
      { alerts?.map((alert, index) => (
        <Alert
          { ...omit(alert, [
            'id', 'type', 'content', 'duration', 'lifespan', 'icon', 'animate',
            'animationTimeout', 'onDismiss',
          ]) }
          key={alert.id ?? index}
          index={alert.id ?? index}
          animate={alert.animate ?? animateAlert}
          animationTimeout={alert.animationTimeout ?? animationTimeout}
          icon={
            alert.icon ?? icons?.[alert.type as AlertsTypes] ?? icons?.default
          }
          lifespan={alert.duration ?? alert.lifespan}
          onDismiss={onDismiss.bind(null, alert)}
          className={alert.type}
        >
          { alert.content }
        </Alert>
      )) }
    </div>
  );
});

Alerts.displayName = 'Alerts';

export default Alerts;
