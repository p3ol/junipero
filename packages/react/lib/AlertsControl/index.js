import {
  forwardRef,
  useCallback,
  useImperativeHandle,
  useReducer,
} from 'react';
import { mockState } from '@junipero/core';
import PropTypes from 'prop-types';

import { AlertsContext } from '../contexts';

const AlertsControl = forwardRef(({ alerts, ...rest }, ref) => {
  const [state, dispatch] = useReducer(mockState, {
    alerts: alerts ?? [],
  });

  useImperativeHandle(ref, () => ({
    alerts: state.alerts,
    add,
    dismiss,
    isJunipero: true,
  }));

  const add = alert => {
    state.alerts.push(alert);
    dispatch({ alerts: state.alerts });
  };

  const dismiss = index => {
    state.alerts.splice(index, 1);
    dispatch({ alerts: state.alerts });
  };

  const getContext = useCallback(() => ({
    alerts: state.alerts,
    add,
    dismiss,
  }), [state.alerts]);

  return (
    <AlertsContext.Provider { ...rest } value={getContext()} />
  );
});

AlertsControl.displayName = 'AlertsControl';
AlertsControl.propTypes = {
  alerts: PropTypes.array,
};

export default AlertsControl;
