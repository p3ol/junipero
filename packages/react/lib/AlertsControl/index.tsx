import {
  ComponentProps,
  ComponentPropsWithRef,
  ComponentPropsWithoutRef,
  ComponentRef,
  ForwardedRef,
  MutableRefObject,
  PropsWithRef,
  forwardRef,
  useCallback,
  useImperativeHandle,
  useState,
} from 'react';
import PropTypes from 'prop-types';

import { AlertsContext, AlertsContextType } from '../contexts';
import { AlertObject } from '../Alert';
import { ForwardedProps } from '../utils';

export declare type AlertsControlRef ={
  alerts: Array<AlertObject>;
  isJunipero: boolean;
  add(alert: AlertObject): void;
  dismiss(alert: AlertObject): void;
} | React.RefAttributes<HTMLElement>;

declare interface AlertsControlProps extends ComponentPropsWithRef<any> {
  alerts?: Array<AlertObject>;
  generateId?: (alert: AlertObject) => string | number;
  ref?: MutableRefObject<AlertsControlRef | undefined>;
  children?: React.ReactNode;
}

const AlertsControl = forwardRef(({
  alerts: alertsProp,
  generateId,
  ...rest
}: AlertsControlProps, ref) => {
  const [alerts, setAlerts] = useState(alertsProp || []);

  useImperativeHandle(ref, () => ({
    alerts,
    add,
    dismiss,
    isJunipero: true,
  }));

  const add = (alert: AlertObject) => {
    alert.id = alert.id ||
      generateId ? generateId(alert) : Math.random().toString(36);

    setAlerts(a => a.concat(alert));
  };

  const dismiss = (alert: AlertObject) => {
    setAlerts(a => a.filter(a => a !== alert));
  };

  const getContext = useCallback<() => AlertsContextType>(() => ({
    alerts,
    add,
    dismiss,
  }), [alerts]);

  return (
    <AlertsContext.Provider { ...rest } value={getContext()} />
  );
}) as ForwardedProps<AlertsControlProps, AlertsControlRef>;

AlertsControl.displayName = 'AlertsControl';
AlertsControl.propTypes = {
  alerts: PropTypes.array,
  generateId: PropTypes.func,
};
export default AlertsControl;
