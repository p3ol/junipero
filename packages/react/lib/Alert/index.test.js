import { render, act } from '@testing-library/react';

import AlertsControl from '../AlertsControl';
import Alerts from '../Alerts';
import Alert from './index';

describe('<Alert />', () => {
  it('should render', () => {
    const { container, unmount } = render(<Alert title="Title">Content</Alert>);
    expect(container).toMatchSnapshot();
    unmount();
  });

  it('should be auto dismissable', async () => {
    jest.useFakeTimers();
    const { container, unmount } = render(
      <AlertsControl
        alerts={[{ title: 'Title', content: 'Content', lifespan: 100 }]}
      >
        <Alerts />
      </AlertsControl>
    );
    expect(container).toMatchSnapshot();

    await act(async () => jest.advanceTimersByTime(100));
    expect(container).toMatchSnapshot();

    unmount();
  });
});
