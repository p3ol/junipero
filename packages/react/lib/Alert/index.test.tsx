import { render, act } from '@testing-library/react';

import AlertsControl from '../AlertsControl';
import Alerts from '../Alerts';
import Alert from './index';

describe('<Alert />', () => {
  it('should render', () => {
    const { container, unmount } = render(<Alert title="Title">Content</Alert>);
    expect(container);
    unmount();
  });

  it('should be auto dismissable', async () => {
    jest.useFakeTimers();
    const alert = { id: 1, title: 'Title', content: 'Content', lifespan: 100 };
    const { container, unmount } = render(
      <AlertsControl alerts={[alert]}>
        <Alerts />
      </AlertsControl>
    );
    expect(container).toMatchSnapshot();

    await act(async () => jest.advanceTimersByTime(100));
    await act(async () => jest.advanceTimersByTime(1));
    expect(container).toMatchSnapshot();

    unmount();
  });
});
