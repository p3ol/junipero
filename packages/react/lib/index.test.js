import { useReducer } from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { mockState } from '@junipero/core';
import { useTimeout } from '@junipero/hooks';

import TextField from './TextField';

describe('mockState: inside react', () => {
  it('should fail update the state correctly and return the old ' +
    'state', async () => {
    const Comp = () => {
      const [state, dispatch] = useReducer(mockState, {
        value: {
          text: 'This is a test',
          content: [],
        },
      });

      useTimeout(() => {
        dispatch({ value: { ...state.value, content: ['foo'] } });
      }, 1, []);

      const onTextChange = ({ value }) => {
        dispatch({ value: { ...state.value, text: value } });
      };

      const onContentChange = () => {
        dispatch({
          value: { ...state.value, content: [...state.value.content, 'bar'] },
        });
      };

      return (
        <div>
          <TextField
            data-testid="Field"
            value={state.value.text}
            onChange={onTextChange}
          />
          <button onClick={onContentChange}>Add content</button>
          <div>Text: { state.value.text }</div>
          <div>Content: { state.value.content.join(', ') }</div>
        </div>
      );
    };

    const { container, unmount } = render(<Comp />);

    fireEvent.change(screen.getByTestId('Field'),
      { target: { value: 'This is a test 2' } });

    await waitFor(() => screen.getByText('Content: foo'));
    expect(container).toMatchSnapshot('With updated text field value');

    fireEvent.click(screen.getByText('Add content'));

    expect(container)
      .toMatchSnapshot('With updated content but with old text value');

    unmount();
  });

  it('should not fail update the state correctly and return the new ' +
    'state when using mockState with a callback', async () => {
    const Comp = () => {
      const [state, dispatch] = useReducer(mockState, {
        value: {
          text: 'This is a test',
          content: [],
        },
      });

      useTimeout(() => {
        dispatch(s => ({ value: { ...s.value, content: ['foo'] } }));
      }, 1, []);

      const onTextChange = ({ value }) => {
        dispatch(s => ({ value: { ...s.value, text: value } }));
      };

      const onContentChange = () => {
        dispatch(s => ({
          value: { ...s.value, content: [...s.value.content, 'bar'] },
        }));
      };

      return (
        <div>
          <TextField
            data-testid="Field"
            value={state.value.text}
            onChange={onTextChange}
          />
          <button onClick={onContentChange}>Add content</button>
          <div>Text: { state.value.text }</div>
          <div>Content: { state.value.content.join(', ') }</div>
        </div>
      );
    };

    const { container, unmount } = render(<Comp />);

    fireEvent.change(screen.getByTestId('Field'),
      { target: { value: 'This is a test 2' } });

    await waitFor(() => screen.getByText('Content: foo'));
    expect(container).toMatchSnapshot('With updated text field value');

    fireEvent.click(screen.getByText('Add content'));

    expect(container)
      .toMatchSnapshot('With updated content AND updated text field valud');

    unmount();
  });
});
