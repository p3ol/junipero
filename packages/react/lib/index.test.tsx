import { useReducer } from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { mockState } from '@junipero/core';
import { useTimeout } from '@junipero/hooks';

import type { FieldContent } from './types';
import TextField from './TextField';

interface State {
  value: {
    text: string;
    content: string[];
  };
}

describe('mockState: inside react', () => {
  it('should correctly update the state and return the new ' +
    'state when using mockState with a callback', async () => {
    const Comp = () => {
      const [state, dispatch] = useReducer(mockState<State>, {
        value: {
          text: 'This is a test',
          content: [],
        },
      });

      useTimeout(() => {
        dispatch(s => ({ value: { ...s.value, content: ['foo'] } }));
      }, 1, []);

      const onTextChange = ({ value }: FieldContent) => {
        dispatch(s => ({ value: { ...s.value, text: value } }));
      };

      const onContentChange = () => {
        dispatch(s => ({
          value: { ...s.value, content: [...s.value.content, 'bar'] },
        }));
      };

      return (
        <div>
          <TextField
            data-testid="Field"
            value={state.value.text}
            onChange={onTextChange}
          />
          <button onClick={onContentChange}>Add content</button>
          <div>Text: { state.value.text }</div>
          <div>Content: { state.value.content.join(', ') }</div>
        </div>
      );
    };

    const { container, unmount } = render(<Comp />);

    fireEvent.change(screen.getByTestId('Field'),
      { target: { value: 'This is a test 2' } });

    await waitFor(() => screen.getByText('Content: foo'));
    expect(container).toMatchSnapshot('With updated text field value');

    fireEvent.click(screen.getByText('Add content'));

    expect(container)
      .toMatchSnapshot('With updated content AND updated text field valud');

    unmount();
  });
});
