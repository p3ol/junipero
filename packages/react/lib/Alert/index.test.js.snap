// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> should be auto dismissable 1`] = `
<div>
  <div
    class="junipero alerts"
  >
    <div
      class="junipero alert"
    >
      <div
        class="junipero card"
      >
        <div
          class="content"
        >
          <h6
            class="junipero"
          >
            Title
          </h6>
          <div
            class="junipero secondary"
          >
            Content
          </div>
        </div>
        <svg
          class="junipero icon remove"
          height="10"
          viewBox="0 0 9 10"
          width="10"
        >
          <path
            d="M8 1.5L1 8.5"
          />
          <path
            d="M1 1.5L8 8.5"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Alert /> should be auto dismissable 2`] = `
<div>
  <div
    class="junipero alerts"
  />
</div>
`;

exports[`<Alert /> should render 1`] = `
<div>
  <div
    class="junipero alert"
  >
    <div
      class="junipero card"
    >
      <div
        class="content"
      >
        <h6
          class="junipero"
        >
          Title
        </h6>
        <div
          class="junipero secondary"
        >
          Content
        </div>
      </div>
      <svg
        class="junipero icon remove"
        height="10"
        viewBox="0 0 9 10"
        width="10"
      >
        <path
          d="M8 1.5L1 8.5"
        />
        <path
          d="M1 1.5L8 8.5"
        />
      </svg>
    </div>
  </div>
</div>
`;
